File: solutions/605_can_place_flowers.ts
function canPlaceFlowers(flowerbed: number[], n: number): boolean {
  for (let i = 0; i < flowerbed.length; i++) {
    if (flowerbed[i]) {
      i += 1;
    } else {
      if (
        flowerbed[i - 1] === 0 &&
        i + 1 < flowerbed.length &&
        flowerbed[i + 1] === 0
      ) {
        flowerbed[i] = 1;
        n--;
        if (n === 0) {
          return true;
        }
      } else {
        return false;
      }
    }
  }

  return n > 0 ? false : true;
}

File: solutions_py/605_can_place_flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """

        for i in range(0, len(flowerbed)):
            if flowerbed[i] != 1:
                emptyLeft = i == 0 or flowerbed[i - 1] == 0
                emptyRight = i == len(flowerbed) - 1 or flowerbed[i + 1] == 0

                if emptyLeft and emptyRight:
                    flowerbed[i] = 1
                    n = n - 1
                    if n == 0:
                        return True
                    i = i + 1

        return False
